### PR TITLE
Add apply cal module

### PR DESCRIPTION
**Pull request by jsdillon**
Here's a simple alternative for applying calibration solutions for IDR2.1. I've also added support for loading npz flags.

I don't think it makes sense to add that functionality to hera_cal.io, since the data available in the .npz files makes it impossible to create a DataContainer without the accompanying miriad file. We can think about whether we want to add support for npz flags in io.load_vis (in which case you also have the .uv) or whether we want to eventually deprecate npz flags or improve them.

I screwed up the history on #201, so I'm starting over here. 

This addresses #197.

### CODE REVIEW

File: hera_cal/apply_cal.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            if new_gains.has_key((i,pol[0])) and new_gains.has_key((j,pol[1])) and (old_gains == None
+                or (old_gains.has_key((i,pol[0])) and old_gains.has_key((j,pol[1])))):
+                    gigj_new = new_gains[(i,pol[0])] * np.conj(new_gains[(j,pol[1])])
+                    if old_gains is not None:
+                        gigj_old = old_gains[(i,pol[0])] * np.conj(old_gains[(j,pol[1])])
+                    else:
+                        gigj_old = np.ones_like(gigj_new)
+                    # update all the data, assuming it wasn't flagged to begin with
+                    if gain_convention == 'divide':
+                        data[(i,j,pol)][~data_flags[(i,j,pol)]] *= (gigj_old / gigj_new)[~data_flags[(i,j,pol)]]
```
Comment (zakiali) [inline hera_cal/apply_cal.py]: This line should be "/= " instead of "*="?

File: hera_cal/apply_cal.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    elif gain_convention == 'multiply':
+                        data[(i,j,pol)][~data_flags[(i,j,pol)]] *= (gigj_new / gigj_old)[~data_flags[(i,j,pol)]]
+                    else:
+                        raise KeyError("gain_convention must be either 'divide' or 'multiply'.")
+                    # update data flags
+                    data_flags[(i,j,pol)][cal_flags[(i, pol[0])]] = True
+                    data_flags[(i,j,pol)][cal_flags[(j, pol[1])]] = True
+            else:
+                data_flags[(i,j,pol)] = np.ones_like(data_flags[(i,j,pol)]) #all flagged
```
Comment (zakiali) [inline hera_cal/apply_cal.py]: isn't this redundant? I think the first if statement takes care if this case.

File: hera_cal/apply_cal.py (hunk 1 above)
Comment (zakiali) [reply hera_cal/apply_cal.py]: This does assume that the conventions of both the old and new gains are the same. Do we want to support them not being the same?

**Comment by jsdillon** [comment]
Yeah, that option seems pretty dangerous to me...to allow for the production of files with some antennas calibrated and some not (but also not flagged) seems like it would create problems.

**Pull request merged**